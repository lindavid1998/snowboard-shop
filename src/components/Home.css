.Home {
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('../images/spring-ski.webp');
	background-size: cover;
	background-position: center;
  width: 100%;
  height: 100%;
  color: white;
}

.overlay {
  padding: 10%;
  display: flex;
  flex-direction: column;
}

.hero-text {
  font-size: 1.4rem;
  font-weight: bold;
}

.hero-subtext {
  font-size: 1.2rem;
  margin-bottom: 3rem;
}

.btn-filled {
  background-color: black;
  padding: 14px;
  color: white;
  font-weight: bold;
  align-self: start;  
  transition: all 0.25s ease-in-out;
}

.btn-filled:hover {
  color: black;
  background-color: white;
  transform: translateY(-5px);
}
