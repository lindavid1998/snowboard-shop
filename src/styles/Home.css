.Home {
  background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('../images/spring-ski.webp');
	background-size: cover;
	background-position: center;
  width: 100%;
  height: 100%;
  color: white;
}

.overlay {
  padding: 10%;
  display: flex;
  flex-direction: column;
}

.hero-text {
  font-size: 1.4rem;
  font-weight: bold;
}

.hero-subtext {
  font-size: 1.2rem;
  margin-bottom: 3rem;
}
