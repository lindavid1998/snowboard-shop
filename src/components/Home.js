import '../styles/Home.css';
import { Link } from 'react-router-dom';

function Home() {
	return (
		<div className='Home'>
			<div className='overlay'>
				<div className='hero-text'>Spring Collection</div>
				<div className='hero-subtext'>
					Keep the party going with gear built for springâ€™s sun, slush, and good
					times.
				</div>
				<Link to='/shop'>
					<button className='btn-translate-on-hover btn-filled btn-medium'>
						Shop Now
					</button>
				</Link>
			</div>
		</div>
	);
}

export default Home;
