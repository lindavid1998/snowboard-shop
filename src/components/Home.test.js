import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import Home from './Home';
import renderer from 'react-test-renderer';

describe('Home', () => {
	test('renders correctly', () => {
		const home = renderer
			.create(
				<Router>
					<Home />
				</Router>
			)
			.toJSON();
		expect(home).toMatchSnapshot();
	});

	test(`routes to shop page when 'shop now' is clicked`, async () => {
		render(
			<Router>
				<Home />
			</Router>
		);

		expect(window.location.pathname).toBe('/');

		const user = userEvent.setup();
		const button = screen.getByRole('button', { name: /shop now/i });
		await act(async () => {
			await user.click(button);
		});
		expect(window.location.pathname).toBe('/shop');
	});
});
