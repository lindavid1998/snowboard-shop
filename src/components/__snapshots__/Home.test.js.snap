// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="Home"
>
  <div
    className="overlay"
  >
    <div
      className="hero-text"
    >
      Spring Collection
    </div>
    <div
      className="hero-subtext"
    >
      Keep the party going with gear built for springâ€™s sun, slush, and good times.
    </div>
    <a
      href="/shop"
      onClick={[Function]}
    >
      <button
        className="btn-filled"
      >
        Shop Now
      </button>
    </a>
  </div>
</div>
`;
